src/Fmt_ast.ml

```diff
@@ -1975,7 +1975,10 @@ and fmt_expression c ?(box = true) ?epi ?eol ?parens ?(indent_wrap = 0) ?ext
                       ?ext:(if first then ext else None)
                       ctx binding
                       ~in_:(fun indent ->
-                        fmt_if_k last (break 1 (-indent) $ fmt "in") )
+                          fmt_if_k last
+                            (fmt_or_k c.conf.end_of_line_in
+                               (fmt " in")
```

Comment:
One thing that should be explicitly tested is what happens if just the ` in` would exceed the margin. There is a question of whether it is better to break before the `in`, as if this option was disabled, or to break the line before the `in`. Analogous choices elsewhere in the formatter opt for allowing the `in` to end up on its own line. Usually that is much simpler to implement, and leads to less dramatic formatting changes when code is edited.

Reply:
This is implemented and tested in `long_let_in`. Some of it is definitely implementation choices (I don't want to add too many different options (break before in: always? when there is no space? never?)

I tried to have some sense of consistency:

In the most common case (when it's not just `in` falling in the margin), the general visual impression of the code is a series of vertically aligned `let`s. In other words, the way the visual presentation relates to the AST is that a certain horizontal spacing is reserved for `let` keywords of a series of let-bindings and for the final expression in which all the bindings are available.

In the rarer occasion of the `in` keyword falling alone in the margin, I attempted to keep that general impression/relationship. This is achieved by placing the `in` on its own line with the same level of indentation as the expression bound by the `let`.

```
let x =
  …
  in
let y =
  …
  in
x + y
```

This is the most consistent approach I found. It keeps the same visual clues for the same types of AST.